src/components/panel/panel.js

```diff
@@ -215,19 +215,69 @@ angular
  * @ngdoc method
  * @name MdPanelRef#open
  * @description
- * If the panel is not visible, opens an already created and configured panel.
+ * Convenience method for attaching and showing the panel.
  *
- * @returns {angular.$q.Promise} A promise that is resolved when the panel is
- * closed.
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * opened.
  */
 
 /**
  * @ngdoc method
  * @name MdPanelRef#close
  * @description
- * If the panel is visible, closes the panel, resolving the promise that is
- * returned from `MdPanelRef#open`. This method destroys the reference to the
- * panel. In order to open the panel again, a new one must be created.
+ * Convenience method for hiding and detaching the panel. This method destroys
+ * the reference to the panel. In order to open the panel again, a new one must
+ * be created.
+ *
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * closed.
+ */
+
+/**
+ * @ngdoc method
+ * @name MdPanelRef#attach
+ * @description
+ * Create the panel elements and attach them to the DOM. The panel will be
+ * hidden by default. Call open() or show() to make the panel visible.
+ *
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * attached.
+ */
+
+/**
+ * @ngdoc method
+ * @name MdPanelRef#detach
+ * @description
+ * Removes the panel from the DOM. If the panel is visible, hide the panel,
+ * then detach.
+ *
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * detached.
+ */
+
+/**
+ * @ngdoc method
+ * @name MdPanelRef#show
```

Comment:
I talked a lot about this with @kara; we thought that having both `open` / `close` and `show` / `hide` is going to lead users to doing the wrong thing or just being confused. So, as an alternative, how about we simply do `open` / `close` as they are now, and then have `addClass` / `removeClass` / `toggleClass` to accomplish the setting and unsetting of `display: none` ?